browser-test/src/applicant/application_version_fast_forward.test.ts

```diff
@@ -486,6 +487,503 @@ test.describe('Application Version Fast-Forward Flow', () => {
+          return programIdV3
+        })
+      })
+
+    // Verify we have a new program version id
+    expect(
+      programIdV2,
+      'Verify Program v2 and Program v3 have the different IDs.',
+    ).not.toEqual(programIdV3)
+    expect(
+      programIdV2,
+      'Verify Program v2 ID is less than Program v3 ID.',
+    ).toBeLessThan(programIdV3)
+
+    // Applicant submits application for program v1; it does not change to program v3
+    await test.step('As applicant - active program v3', async () => {
+      await applicantActor.getPage().reload()
+
+      await test.step('As applicant - submit application', async () => {
+        await applicantActor.continueAnswerApplicationQuestions()
+        await applicantActor.answerQuestions([Question.H])
+        await applicantActor.submitApplication()
+      })
+
+      await test.step('As applicant - check program list has one submitted application for program v3', async () => {
+        await applicantActor.gotoApplicantHomePage()
+        await expect(
+          applicantActor.getCardListLocator(
+            ApplicationStatusCardGroupName.InProgress,
+          ),
+        ).not.toBeAttached()
+        await expect(
+          applicantActor.getCardListLocator(
+            ApplicationStatusCardGroupName.Submitted,
+          ),
+        ).toBeAttached()
+
+        const headingLocator = applicantActor.getCardHeadingLocator(
+          ApplicationStatusCardGroupName.Submitted,
+        )
+        await expect(headingLocator).toBeAttached()
+
+        // Once submitted the program id on the edit button will be for the latest version
+        expect(
+          await applicantActor.getProgramIdFromLocator(headingLocator),
+        ).toBe(programIdV3)
+      })
+
+      await test.step('check program list shows eligible tag', async () => {
+        await expect(
+          applicantActor.getCardEligibleTagLocator(
+            ApplicationStatusCardGroupName.Submitted,
+          ),
+        ).toBeVisible()
+
+        await expect(
+          applicantActor.getCardNotEligibleTagLocator(
+            ApplicationStatusCardGroupName.Submitted,
+          ),
+        ).not.toBeAttached()
+      })
+    })
+
+    // Program Admin can view applications submitted for program v3
+    await test.step('As program admin - active program v3', async () => {
+      await test.step('Log in program admin', async () => {
+        // Log in Program Admin here and not at the start. If logged in before the program is created by
+        // a civiform admin, the program admin won't have permissions to the program.
+        await programAdminActor.login()
+        await programAdminActor.viewApplications()
+      })
+
+      await test.step('sees submitted application with questions from program version v3', async () => {
+        const cardLocator = programAdminActor.getCardLocator()
+        await expect(cardLocator).toHaveCount(1)
+
+        const cardButton = cardLocator.getByRole('link', {name: 'View'})
```

Comment:
Question: Is it important to check the ID, or is the purpose of this step to verify that the correct version is showing?

An alternative might be to change the program name for each version and assert the appropriate text is visible/hidden.

Reply:
I want to explicitly verify that the version id is the expected value since it changes under the covers for the applicant. I don't want to rely on just changing the text in case other things affect that and checking by ID maintains a consistent value as I mutate things.